File: 05_arrays/05_06_rotate_array_by_one_place/rotate_array_by_one_place.js
function solve(arr) {
    const temp = arr[0];
    for (let i = 0; i < arr.length - 1; i++) {
        arr[i] = arr[i + 1];
    }
    arr[arr.length - 1] = temp;
}

const arr = [1, 2, 3, 4, 5];
console.log("Elements:-");
console.log(arr);
solve(arr);
console.log("Rotated by one place:-");
console.log(arr);

File: 05_arrays/05_06_rotate_array_by_one_place/RotateArrayByOnePlace.java
public class RotateArrayByOnePlace {
    public static void main(String[] args) {
        int arr[] = { 1, 2, 3, 4, 5 };
        System.out.println("Elements:-");
        print(arr);
        solve(arr);
        System.out.println("Rotated by one place:-");
        print(arr);
    }

    static void solve(int arr[]) {
        int temp = arr[0];
        for (int i = 0; i < arr.length - 1; i++) {
            arr[i] = arr[i + 1];
        }
        arr[arr.length - 1] = temp;
    }

    static void print(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}
